src/packet/key/public.rs

```diff
@@ -394,7 +514,27 @@ impl PublicKeyTrait for PubKeyInner {
         self.to_writer(&mut key_buf)?;
 
         // old style packet header for the key
-        writer.write_all(&[0x99, (key_buf.len() >> 8) as u8, key_buf.len() as u8])?;
+        match self.version() {
+            KeyVersion::V2 | KeyVersion::V3 | KeyVersion::V4 => {
+                // When a v4 signature is made over a key, the hash data starts with the octet 0x99,
+                // followed by a two-octet length of the key, and then the body of the key packet.
+                writer.write_all(&[0x99])?;
```

Comment:
isn't there `write_u8`?

Reply:
I've made a branch that uses write_u8 a lot, starting from the code in this PR: https://github.com/hko-s/rpgp/pull/2

My suggestion would be that we merge that after this one.